### PR TITLE
[INLONG-10402][Audit] Audit Service supports the hourly Audit data one day ago

### LINKED ISSUE #10402: [Improve][Audit]  Audit Service supports the hourly Audit data one day ago
### Description

Audit Service supports hourly the Audit data one day ago

### InLong Component

InLong Audit

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by doleyzi**
- Fixes #10402

### Motivation

Audit Service supports the hourly Audit data one day ago.

### Modifications

Supports querying historical data that does not exist in the cache from the audit storage and automatic partition management.

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]

**Review by justinwwhuang — APPROVED** [review]

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/AbstractCache.java [hunk 1]
```diff
@@ -96,38 +95,47 @@ public Cache<String, StatData> getCache() {
...
-            SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT);
-            Date startDate = dateFormat.parse(startTime);
-            Date endDate = dateFormat.parse(endTime);
-            for (int index = 0; index < MAX_CACHE_KEY_SIZE; index++) {
-                Calendar calendar = Calendar.getInstance();
-                calendar.setTime(startDate);
-                calendar.add(Calendar.MINUTE, index * auditCycle.getValue());
-                calendar.set(Calendar.SECOND, 0);
-                if (calendar.getTime().compareTo(endDate) > 0) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/AbstractCache.java:123]: DateTimeFormatter initialized once is enough.

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java [hunk 2]
```diff
@@ -153,6 +174,46 @@ protected void createDataSource() {
         dataSource = new HikariDataSource(config);
     }
 
+    private void managePartitions() {
+        addPartition();
+        deletePartition();
+    }
+
+    private String formatPartitionName(LocalDate date) {
+        return "p" + date.format(DateTimeFormatter.ofPattern("yyyyMMdd"));
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java:183]: ditto

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java [hunk 3]
```diff
@@ -153,6 +174,46 @@ protected void createDataSource() {
...
+    }
+
+    private void addPartition() {
+        String partitionName = formatPartitionName(LocalDate.now().plusDays(1));
+        String partitionValue = LocalDate.now().plusDays(2).format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java:188]: ditto

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/AbstractCache.java (hunk 1 above)
Comment (doleyzi) [reply inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/AbstractCache.java]: Fixed

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java (hunk 2 above)
Comment (doleyzi) [reply inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java]: Fixed

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java (hunk 3 above)
Comment (doleyzi) [reply inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java]: Fixed

**Review by dockerzhang — APPROVED** [review]

**Review by justinwwhuang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**